File: GCD/Gcd.py
def max_gcd(v):
    if len(v) < 2:
        return 0
    ans = -10**9
    for i in range(1, len(v)):
        a = v[i-1]
        b = v[i]
        if b[0] == a[0]:
            ans = max(ans, gcd(a[1], b[1]))
    return ans

def gcd(n1, n2):
    small = n1 if n1 < n2 else n2
    for i in range(small, 0, -1):
        if n1 % i == 0 and n2 % i == 0:
            return i

if __name__ == "__main__":
    v = [[5, 4], [5, 8], [4, 6], [4, 9], [8, 10], [10, 20], [10, 30]]
    print(max_gcd(v))
    print(gcd(4, 14))

File: GCD/Gcd.java
package GCD;

import java.util.Scanner;

public class Gcd {
    public static void main(String[] args) {
        
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter two numbers:- ");
        int a = sc.nextInt();
        int b = sc.nextInt();
        int gcd = gcd(a, b);
        System.out.println("GCD: " + gcd);
        // int maxGcd = maxGcd(a, b);
        // System.out.println("Max GCD: " + maxGcd);

        sc.close();
    }

    public static int gcd(int a, int b) {
        if (b == 0) {
            return a;
        }
        return gcd(b, a % b);
    }

    // public static int maxGcd(int a, int b) {
    //     int max = 1;
    //     int min = Math.min(a, b);
    //     for (int i = 2; i <= min; i++) {
    //         if (a % i == 0 && b % i == 0) {
    //             max = i;
    //         }
    //     }
    //     return max;
    // }
}



